src/libostree/ostree-sysroot.c

```diff
@@ -1768,9 +1784,9 @@ ostree_sysroot_deployment_unlock (OstreeSysroot     *self,
          * directly for hotfixes.  The ostree-prepare-root.c helper
          * is also set up to detect and mount these.
          */
-        if (!glnx_shutil_mkdir_p_at (deployment_dfd, ".usr-ovl-upper", 0755, cancellable, error))
+        if (!mkdir_unmasked (deployment_dfd, ".usr-ovl-upper", 0755, cancellable, error))
```

Comment:
Optional tweak - inherit the mode of the underlying `/usr` rather than hardcoding it.

Reply:
Good idea, done! :arrow_down: 

(Made this a separate commit to be explicit).